crates/admin/src/service.rs

```diff
@@ -85,24 +85,25 @@ where
                     )),
             );
 
-        // Bind and serve
-        let server = hyper::Server::try_bind(&opts.bind_address)
+        // run our app with hyper
+        let listener = tokio::net::TcpListener::bind(&opts.bind_address)
+            .await
             .map_err(|err| Error::Binding {
                 address: opts.bind_address,
-                source: err,
-            })?
-            .serve(router.into_make_service());
-
-        info!(
-            net.host.addr = %server.local_addr().ip(),
-            net.host.port = %server.local_addr().port(),
-            "Admin API listening"
-        );
-
-        // Wait server graceful shutdown
-        Ok(server
+                source: Box::new(err),
+            })?;
+        if let Ok(local_addr) = listener.local_addr() {
+            info!(
+                net.host.addr = %local_addr.ip(),
+                net.host.port = %local_addr.port(),
+                "Admin API listening"
+            );
+        } else {
+            info!("Admin API listening");
+        }
```

Comment:
Under which condition can the error case happen here and why is it ok to ignore it?

Reply:
I tried to trace it back, and I'm not sure, but I suspect this happens with unix domain sockets...

Reply:
Should we then add an `expect` stating that this should never fail because we provide a `SocketAddr`?

Reply:
I don't think so, how about we want to let people bind to unix domain sockets at some point?

Reply:
Then we probably wouldn't create a `TcpListener` for this case.

Reply:
That's also true... Changing to expect